### PR TITLE
CT-1671 reintroduce regex for filtering out business units

### PR DESCRIPTION
**Pull request by aliuk2012**

### CODE REVIEW

**Review by beccapearce — APPROVED** [review]

**Review by stephenrichards — DISMISSED** [review]
We need to make this work using a search button to the right of the text box where we  enter the text to match the business unit name on. - as per Mia's design.

File: app/assets/javascripts/modules/AssignedToFilter.js [hunk 1]
```diff
@@ -9,29 +9,39 @@ moj.Modules.AssignedToFilter = {
...
-              $elem.find(':checkbox').prop('checked', false);
-          }
-        })
+      .each(function (index, element) {
+        var $elem = $(this);
+        // Remove any character that is not alphanumeric
+        var labelText = $elem.text().replace(/[^a-z0-9+]+/gi, '');
+        // Remove any character that is not alphanumeric
+        var cleanedTerm = searchFor.replace(/[^a-z0-9+]+/gi, '');
+        var check = new RegExp(cleanedTerm, 'gi');
```
Comment (misaka) [inline app/assets/javascripts/modules/AssignedToFilter.js]: These two lines don't have to happen for every element in `$('.js-all-business-units').find('.multiple-choice')` does it?

File: app/assets/javascripts/modules/AssignedToFilter.js [hunk 2]
```diff
@@ -9,29 +9,39 @@ moj.Modules.AssignedToFilter = {
...
+
+
+        // if checkbox label test contains any part of the search term
+        if (check.test(labelText)) {
+          $elem.show();
+        } else {
+          $elem.hide();
+          $elem.find(':checkbox').prop('checked', false);
```
Comment (misaka) [inline app/assets/javascripts/modules/AssignedToFilter.js]: Shouldn't all the business units that have been previously selected be replayed?

Comment (misaka) [reply app/assets/javascripts/modules/AssignedToFilter.js]: As per Slack discussion, we'll leave this for a later improvement and just get this out for now.

**Review by misaka — APPROVED** [review]

**Pull request merged**